src/Appwrite/Auth/OAuth2/Exception.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Extend\Exception as AppwriteException;
+
+class Exception extends AppwriteException
+{
+    protected string $response = '';
+    protected string $error = '';
+    protected string $errorDescription = '';
+
+    public function __construct(string $response = '', int $code = 0, \Throwable $previous = null)
+    {
+        $this->response = $response;
+        $this->message = $response;
+        $decoded = json_decode($response, true);
+        if (\is_array($decoded)) {
+            $this->error = $decoded['error'];
+            $this->errorDescription = $decoded['error_description'];
+            $this->message =  $this->error . ': ' . $this->errorDescription;
+        }
+        $type = match($code) {
+            400 => AppwriteException::USER_OAUTH2_BAD_REQUEST,
+            401 => AppwriteException::USER_OAUTH2_UNAUTHORIZED,
+            default => AppwriteException::GENERAL_SERVER_ERROR
+        };
+
+        parent::__construct($type, $this->message, $code, $previous);
+    }
+
+    public function getError() : string
+    {
+        return $this->error;
```

Comment:
Why aren't we using the native getMessage interface?

Reply:
This is the error from the oauth2 provider. The message will include error and error_description.

See https://datatracker.ietf.org/doc/html/rfc6749#section-5.2